src/server/game/Anticheat/AnticheatMgr.cpp

```diff
@@ -15,82 +17,166 @@
 {
-    if ((sWorld->getIntConfig(CONFIG_ANTICHEAT_DETECTIONS_ENABLED) & JUMP_HACK_DETECTION) == 0)
+    if (!sWorld->getBoolConfig(CONFIG_ANTICHEAT_JUMPHACK_ENABLE))
         return;
 
+    // we pull the player's individual guid
     uint32 key = player->GetGUID().GetCounter();
 
+    // Chain or double multi jumping is not a thing in 335
     if (m_Players[key].GetLastOpcode() == MSG_MOVE_JUMP && opcode == MSG_MOVE_JUMP)
     {
-        BuildReport(player,JUMP_HACK_REPORT);
-        TC_LOG_DEBUG("entities.player.character", "AnticheatMgr:: Jump-Hack detected player GUID %s",player->GetGUID().ToString().c_str());
+        if (sWorld->getBoolConfig(CONFIG_ANTICHEAT_WRITELOG_ENABLE))
+        {
+            TC_LOG_INFO("anticheat", "AnticheatMgr:: Jump-Hack detected player %s (%s)", player->GetName().c_str(), player->GetGUID().ToString().c_str());
+        }
+        BuildReport(player, JUMP_HACK_REPORT);
     }
 }
 
-void AnticheatMgr::WalkOnWaterHackDetection(Player* player, MovementInfo /* movementInfo */)
+void AnticheatMgr::WalkOnWaterHackDetection(Player* player, MovementInfo movementInfo)
 {
-    if ((sWorld->getIntConfig(CONFIG_ANTICHEAT_DETECTIONS_ENABLED) & WALK_WATER_HACK_DETECTION) == 0)
+    if (!sWorld->getBoolConfig(CONFIG_ANTICHEAT_WATERWALKHACK_ENABLE))
         return;
 
+    // we pull the player's individual guid
     uint32 key = player->GetGUID().GetCounter();
-    if (!m_Players[key].GetLastMovementInfo().HasMovementFlag(MOVEMENTFLAG_WATERWALKING))
+    uint32 distance2D = (uint32)movementInfo.pos.GetExactDist2d(&m_Players[key].GetLastMovementInfo().pos);
+
+    // We don't need to check for a waterwalk hack if the player hasn't moved
+    // This is necessary since MovementHandler fires if you rotate the camera in place
+    if (!distance2D)
         return;
 
+    // if the player is water walking on water then we are good.
+    if (player->GetLiquidStatus() == LIQUID_MAP_WATER_WALK)
+    {
+        if (!m_Players[key].GetLastMovementInfo().HasMovementFlag(MOVEMENTFLAG_WATERWALKING) && !movementInfo.HasMovementFlag(MOVEMENTFLAG_WATERWALKING))
+        {
+            if (sWorld->getBoolConfig(CONFIG_ANTICHEAT_WRITELOG_ENABLE))
+            {
+                TC_LOG_INFO("anticheat", "AnticheatMgr:: Walk on Water - Hack detected player %s (%s)", player->GetName().c_str(), player->GetGUID().ToString().c_str());
+            }
+            BuildReport(player, WALK_WATER_HACK_REPORT);
+        }
+    }
+
     // if we are a ghost we can walk on water
     if (!player->IsAlive())
         return;
 
-    if (player->HasAuraType(SPELL_AURA_FEATHER_FALL) ||
-        player->HasAuraType(SPELL_AURA_SAFE_FALL) ||
-        player->HasAuraType(SPELL_AURA_WATER_WALK))
+    // Prevents the False Positive for water walking when you ressurrect.
+    // Aura 15007 (Resurrection sickness) is given while dead before returning back to life.
+    if (player->HasAuraType(SPELL_AURA_GHOST) && player->HasAura(15007))
```

Comment:
please move the magic number 15007 to a define or enum or something with a label

Reply:
https://github.com/TrinityCore/TrinityCoreCustomChanges/pull/73/commits/7441be6078e8b03adf999d1c52a7be8dccd8b89d